application/web/web_main.py

```
@@ -306,107 +678,120 @@ def find_document_by_tag() -> Any:
+            if isinstance(existing_standards, (list, tuple, set)):
+                existing_lower = {str(s).lower() for s in existing_standards}
+                missing = [s for s in standards if str(s).lower() not in existing_lower]
+                if missing:
+                    logger.info(
+                        f"On Heroku: gap analysis request {standards_hash} references "
+                        f"standards that do not exist: {', '.join(missing)}, returning 404"
+                    )
+                    abort(
+                        404, f"One or more standards do not exist: {', '.join(missing)}"
+                    )
+        except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)